### DISCUSSION TITLE
Add custom data to __NEXT_DATA__

### DISCUSSION THREAD

**Discussion by hildebert** [vercel/next.js#12074] [Help]
I'm working on making existing python (Django) app to work together with NextJS+SSR. I'm sending requests from Django to NextJS server and rendering response. 

I would like to be able to pass some context data from Django to NextJS like, for example, user info and then access this data in NextJS components.

Most natural way for me would be to add this data to "__NEXT_DATA__", however I'm having hard time finding a way to do it. I manager to pass it to __NEXT_DATA__.query, however it doesn't feel right.

What is the proper way to make data available inside NextJS components?

**Comment by Wounded-Knee** [comment]
Same question.

**Comment by stigkj** [comment]
Not sure I understand how this is setup. Do you have the Django app before the Next.js app, i.e. the requests hit the Django app which then calls the Next.js app? Then what do you do with the result from Next.js? Inject it into a `<div>` or something? Not sure that would work very well.

In regards to passing some context data to Next.js, I would think just passing URL parameters would suffice?
